internal/bminventory/inventory.go

```diff
@@ -1315,11 +1316,73 @@ func (b *bareMetalInventory) TransformClusterToDay2Internal(ctx context.Context,
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	log.Infof("Pre-update TransformClusterToDay2Internal APIVIP %v", cluster.APIVip)
+	log.Infof("Pre-update TransformClusterToDay2Internal APIVIPDNSName %v", cluster.APIVipDNSName)
+	if cluster.IgnitionEndpoint == nil || cluster.IgnitionEndpoint.URL == nil {
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		log.Infof("Empty IgnitionEndpoint found")
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), models.HostRoleWorker)
+		params := installer.V2UpdateClusterParams{
+			ClusterUpdateParams: &models.V2ClusterUpdateParams{
+				IgnitionEndpoint: &models.IgnitionEndpoint{
+					URL: &ignitionEndpoint,
+				},
+			},
+			ClusterID: *cluster.ID,
+		}
+
+		usages, errUsages := usage.Unmarshal(cluster.Cluster.FeatureUsage)
+		if errUsages != nil {
+			log.WithError(errUsages).Errorf("failed to read feature usage from cluster %s", params.ClusterID)
+			return nil, errUsages
+		}
+
+		txSuccess := false
+		tx := b.db.Begin()
+		defer func() {
+			if !txSuccess {
+				log.Error("update cluster failed")
+				tx.Rollback()
+			}
+			if r := recover(); r != nil {
+				log.Errorf("update cluster failed to recover: %s", r)
+				log.Error(string(debug.Stack()))
+				tx.Rollback()
+			}
+		}()
+
+		if errClusterUpdate := b.updateClusterData(ctx, cluster, params, usages, tx, log, Interactive); errClusterUpdate != nil {
```

Comment:
Why not just use ClusterUpdateInternal. Not sure why we need to have this copy/paste logic in 2 different functions

Reply:
ClusterUpdateInternal exits early when it noticed that cluster has completed installation